datumaro/plugins/dataset_validator.py

```diff
@@ -0,0 +1,733 @@
+# Copyright (C) 2021 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import argparse
+import json
+import logging as log
+import numpy as np
+from copy import deepcopy
+
+from datumaro.components.validator import (Severity, TaskType, Validator)
+from datumaro.components.cli_plugin import CliPlugin
+from datumaro.components.dataset import IDataset
+from datumaro.components.errors import MultiLabelAnnotations, NegativeLength
+from datumaro.components.extractor import AnnotationType, LabelCategories
+from datumaro.util import parse_str_enum_value
+
+from datumaro.cli.util import MultilineFormatter
+from datumaro.cli.util.project import generate_next_file_name, load_project
+
+class DatasetValidator(Validator, CliPlugin):
```

Comment:
I still can't understand the purpose of this class.

If it provides some shared bits of implementation - make it a base class for others. If it provides a common interface - put that code into the base class. Now it violates the interface, which it is supposed to implement, and can't be used the same way as other Validator classes. 

I see it the same way as extractors defined: 
IExtractor - abstract interface, no implementations (not necessary)
Extractor - some common and default definitions of methods
SourceExtractor - a base class for an often scenario
format extractors - which are format specific and inherit either Extractor or SourceExtractor

Here is the same:
Validator - a common interface or a base class
_TaskValidator - a base class for this kind of validators
Classification etc. Validator - specific classes.

They are used this way:
```
validator_type = env.validators[task]
extra_args = validator_type.parse_cmdline(...)
validator = validator_type(extra_args)
report = validator.validate(dataset)
```

Reply:
I applied your feedback.

Reply:
Hi Maxim,
Thanks for your feedbacks. I applied all again. Please review it again.